src/views/List.jsx

```diff
@@ -1,57 +1,81 @@
 import { ListItem } from '../components';
 import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 export function List({ data }) {
-	const [searchInput, setSearchInput] = useState('');
+	const navigate = useNavigate();
 
-	const listItemsToDisplay = data.map((item) => {
-		const isItemInSearch = item.name
-			?.toLowerCase()
-			.includes(searchInput.toLowerCase());
+	const WelcomePrompt = () => {
+		return (
+			<>
+				<h3>Shopping list is currently empty...</h3>
+				<button
+					type="button"
+					onClick={() => {
+						navigate('/add-item');
+					}}
+				>
+					Add Item
+				</button>
+			</>
+		);
+	};
 
-		return isItemInSearch ? <ListItem key={item.id} name={item.name} /> : null;
-	});
+	const FormAndList = () => {
+		const [searchInput, setSearchInput] = useState('');
 
-	const handleKeyDown = (e) => {
-		if (e.keyCode === 13) {
+		const handleKeyDown = (e) => {
+			if (e.keyCode === 13) {
+				e.preventDefault();
+			}
+		};
+		const resetDisplayList = (e) => {
 			e.preventDefault();
-		}
-	};
+			setSearchInput('');
+		};
+
+		const listItemsToDisplay = data.map((item) => {
+			const isItemInSearch = item.name
+				?.toLowerCase()
+				.includes(searchInput.toLowerCase());
 
-	const resetDisplayList = (e) => {
-		e.preventDefault();
-		setSearchInput('');
+			return isItemInSearch ? (
+				<ListItem key={item.id} name={item.name} />
+			) : null;
+		});
+
+		return (
+			<>
+				<form>
+					<label htmlFor="filter">Filter List</label>
+					<br />
+					<input
+						type="text"
+						name="filter"
+						id="filter"
+						value={searchInput}
+						placeholder="Item name"
+						onChange={(e) => {
+							setSearchInput(e.target.value);
+						}}
+						onKeyDown={(e) => {
+							handleKeyDown(e);
+						}}
+					></input>
+					<button
+						type="button"
+						name="clearInput"
+						onClick={(e) => {
+							resetDisplayList(e);
+						}}
+					>
+						X
+					</button>
+				</form>
+				<ul>{listItemsToDisplay}</ul>
+			</>
+		);
 	};
 
-	return (
-		<>
-			<form>
-				<label htmlFor="filter">Filter List</label>
-				<br />
-				<input
-					type="text"
-					name="filter"
-					id="filter"
```

Comment:
Thank you for matching htmlFor attribute in labels and id to make it more accessible.